packages/lys/src/slice.ts

```diff
@@ -1,8 +1,19 @@
 import { createDraft, Draft, finishDraft } from "immer";
 import { ObjectPatcher, patchObject } from "./patchObject";
+import { DeepReadonly } from "./typeutils";
 
 export type SliceDefinition<State> = {
-  [K: string]: SliceAction<State>;
+  [K: string]:
+    | SliceAction<State>
+    | { [K: string]: SliceSelector<State> }
+    | undefined;
+  selectors?: {
```

Comment:
I'm not sure if Selector is the right name, but the intention of this feature is to describe a sub-state that can be definitively determined from the state...

Reply:
Redux has a lot of states to deal with, so the name Selector was the right name for the state extraction layer from the Store, but Lys doesn't want to extract states, it wants to get states without the need to declare them procedurally.